opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/__init__.py

```diff
@@ -543,9 +543,10 @@ def emit(self, record: logging.LogRecord) -> None:
 
     def flush(self) -> None:
         """
-        Flushes the logging output.
+        Flushes the logging output. Skip flushing if logger is NoOp.
         """
-        self._logger_provider.force_flush()
+        if not isinstance(self._logger, NoOpLogger):
```

Comment:
It would be more pythonic to call the method here and ignore the exception if it doesn't exist, and/or implement force_flush on NoOpLogger so it doesn't error

Reply:
I agree! I tried to follow the same style of emit method, but I think we could just check if the attribute exist. I don't think the noop should implement these methods to silence the problem, because the `Logger` ABC don't expect them.

Reply:
Nice, yeah the first one will survive refactors.

I'm not affiliated with the project, just somebody getting the same error :)

Reply:
Any reason not to define `force_flush` in the `LoggerProvider` abstract base class and implementing it as a no-op in `NoOpLoggerProvider`. Then there will be no need for the `isinstance` or `has_attr`.

Reply:
I don't think all providers need a force flush, but this is already required today, only the noop provider don't have it. I'll change it.

Reply:
We expect that abstract methods will be implemented (`NoOpLogger` is a special case) and this is true for all other abstract methods in three signals.